testhelper/cleanup.go

```diff
@@ -1,20 +1,22 @@
 package testhelper
 
-import "log"
+import "fmt"
 
 type Cleanup struct {
-	fns []func() error
+	fns []func()
 }
 
-func (c *Cleanup) Defer(fn func() error) {
+func (*Cleanup) Log(a ...interface{}) {
+	fmt.Println(a...)
+}
+
+func (c *Cleanup) Cleanup(fn func()) {
```

Comment:
Suggestion:

`func (c * Cleanup) Push (fn func()) {`

Reply:
I named it `Cleanup` for a specific reason, see PR description:

![Screenshot from 2022-06-17 13-38-37](https://user-images.githubusercontent.com/1288256/174291222-c851fa14-fc36-444d-b8fd-6bad1ffbd282.png)

This way, if we are in a test we can just pass `t *testing.T` instead of using the `Cleanup` struct. Ideally we should get to a point where we're not using the struct and we can remove it but we're still discussing it as it's part of the "avoid TestMain" discussion.